columnflow/tasks/ml.py

```diff
@@ -51,6 +54,9 @@ class PrepareMLEvents(
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
+        # cache for producer inst
+        self._preparation_producer_inst = None
```

Comment:
Maybe use `law.no_value` here instead of *None* to mark this attribute as "not cached so far", since *None* is a viable value in case no preparation producer is configured.